### PR TITLE
Add world border checking to teleportation safety

### LINKED ISSUE #3499: TPR can teleport outside worldborder
When using TPR on the latest version of EssentialsX and Paper, I was teleported outside the worldborder far enough to where I was instantly killed and lost all my items as there was no way to retrieve them. I do believe this is an issue as that does not feel 'safe'.

### PR DESCRIPTION
**Pull request by pop4959**
Fixes #3499. Please see that issue for discussion about the bug.

Fix demonstration with `/tppos`: https://imgur.com/a/Mo4okQa

### CODE REVIEW

**Review by JRoy — APPROVED** [review]

File: Essentials/src/com/earth2me/essentials/utils/LocationUtil.java [hunk 1]
```diff
@@ -175,6 +186,22 @@ public static Location getSafeDestination(final Location loc) throws Exception {
...
+                x = x1;
+            } else if (x > x2) {
+                x = x2;
+            }
+            if (z < z1) {
+                z = z1;
+            } else if (z > z2) {
+                z = z2;
+            }
+        }
```
Comment (JRoy) [inline Essentials/src/com/earth2me/essentials/utils/LocationUtil.java]: After further review, this code should be placed after this line in AsyncTeleport so we do not create extraneous sync loads https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/com/earth2me/essentials/AsyncTeleport.java#L181

**Review by JRoy — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/utils/LocationUtil.java (hunk 1 above)
Comment (pop4959) [reply Essentials/src/com/earth2me/essentials/utils/LocationUtil.java]: I still don't think this causes any sync loads. It's just getting the position of the block, not the block itself. Reducing sync loads in LocationUtil would probably be best for a separate PR, as that is not the concern of this one.

**Review by JRoy — CHANGES_REQUESTED** [review]
looks fine? config option tho? :)

**Comment by pop4959** [comment]
> looks fine? config option tho? :)

<https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/config.yml#L75>
<https://github.com/EssentialsX/Essentials/pull/3536/files#diff-9e1c35687809245f3b34cee2707f2988R183>

It's disabled by `teleportation-safety`, and also doesn't trigger if the player is invulnerable (god mode, creative, etc).

**Review by JRoy — APPROVED** [review]
lol ok

**Review by mdcfe — APPROVED** [review]

**Pull request merged**